tests/components/nest/test_config_flow_sdm.py

```diff
@@ -0,0 +1,66 @@
+"""Test the Google Nest Device Access config flow."""
+from homeassistant import config_entries, setup
+from homeassistant.components.nest.const import DOMAIN, OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from tests.async_mock import patch
+
+CLIENT_ID = "1234"
+CLIENT_SECRET = "5678"
+PROJECT_ID = "project-id-4321"
+SUBSCRIBER_ID = "subscriber-id-9876"
+
+
+async def test_full_flow(hass, aiohttp_client, aioclient_mock, current_request):
+    """Check full flow."""
+    assert await setup.async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {
+                "project_id": PROJECT_ID,
+                "subscriber_id": SUBSCRIBER_ID,
+                CONF_CLIENT_ID: CLIENT_ID,
+                CONF_CLIENT_SECRET: CLIENT_SECRET,
+            },
+            "http": {"base_url": "https://example.com"},
+        },
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(hass, {"flow_id": result["flow_id"]})
+
+    oauth_authorize = OAUTH2_AUTHORIZE.format(project_id=PROJECT_ID)
+    assert result["url"] == (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)